chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -310,6 +313,14 @@ impl PeerManagerActor {
                             },
                         )
                     }
+                    // After we add or remove local peer, we should update routing table imediately.
+                    if update_routing_table_now {
+                        act.update_routing_table_and_prune_edges(
```

Comment:
so this issue was not caught by any tests? 

can you add a test that would cover this?

Reply:
It was, it was causing some random test failures.

Reply:
Usually, we update routing table after 1s. However, if local edge gets updated, we can do it immediately to avoid up to 1s delay.